### PR TITLE
Fixing the deletion flow for files and folders

### PR DESCRIPTION
**Pull request by Benjin**
This also adds "entryKey" to tree file nodes, which is the key by which DacFx/STS does lookups for Contain/Get/Delete/Exclude.

### CODE REVIEW

File: extensions/sql-database-projects/src/models/tree/projectTreeItem.ts [hunk 1]
```diff
@@ -67,55 +67,54 @@ export class ProjectRootTreeItem extends BaseProjectTreeItem {
 	 * Processes the list of files in a project file to constructs the tree
 	 */
 	private construct() {
+		// folders
+		for (const folder of this.project.folders) {
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/tree/projectTreeItem.ts:71]: is this needed? The folders are also getting added below on line 117

File: extensions/sql-database-projects/src/models/tree/projectTreeItem.ts [hunk 2]
```diff
@@ -145,23 +144,23 @@ export class ProjectRootTreeItem extends BaseProjectTreeItem {
...
-		let current: fileTree.FolderNode | ProjectRootTreeItem = this;
+		let current: fileTree.FolderNode | ProjectRootTreeItem = this; // start with the Project root node
 
-		for (const part of relativePathParts) {
-			if (current.fileChildren[part] === undefined) {
+		for (const part of relativePathParts) { // iterate from the project root, down the path to the entry in question
+			if (current.fileChildren[part] === undefined) { // if the current node doesn't have the next child node in its list of children, add it
 				const parentPath = current instanceof ProjectRootTreeItem ? path.dirname(current.fileSystemUri.fsPath) : current.fileSystemUri.fsPath;
-				current.fileChildren[part] = new fileTree.FolderNode(vscode.Uri.file(path.join(parentPath, part)), this.projectFileUri);
+				current.fileChildren[part] = new fileTree.FolderNode(vscode.Uri.file(path.join(parentPath, part)), this.projectFileUri, 'shouldnt happen');
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/tree/projectTreeItem.ts:156]: was the `'shouldnt happen'` for debugging?

File: extensions/sql-database-projects/src/models/tree/sqlcmdVariableTreeItem.ts [hunk 3]
```diff
@@ -64,6 +64,7 @@ export class SqlCmdVariablesTreeItem extends BaseProjectTreeItem {
 export class SqlCmdVariableTreeItem extends BaseProjectTreeItem {
 	constructor(private sqlcmdVar: string, sqlCmdNodeRelativeProjectUri: vscode.Uri, sqlprojUri: vscode.Uri,) {
 		super(vscode.Uri.file(path.join(sqlCmdNodeRelativeProjectUri.fsPath, sqlcmdVar)), sqlprojUri);
+		this.entryKey = this.friendlyName;
```
Comment (kisantia) [inline extensions/sql-database-projects/src/models/tree/sqlcmdVariableTreeItem.ts:67]: should database reference tree item also have `entryKey` set so all the project's child nodes that can be added/deleted have it set?

File: extensions/sql-database-projects/src/models/tree/projectTreeItem.ts (hunk 1 above)
Comment (Benjin) [reply extensions/sql-database-projects/src/models/tree/projectTreeItem.ts:71]: Ah, I was moving the folder reading up here, and 117 must've been re-added during a merge.

**Review by kisantia — APPROVED** [review]

**Pull request merged**